### PR TITLE
New node setup

### PR DESCRIPTION
**Pull request by gothrek22**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,69 +3,75 @@ on:
   push:
     branches:
       - master
-  pull_request:
+      - feature/new_node_setup
   schedule:
-    - cron: "0 2 * * 0"
+    - cron: 0 2 * * 0
+permissions:
```
Comment (mb-shorai) [inline .github/workflows/main.yml:8]: The only required things here are probably contents (RO) and packages (RW). All other permissions will be automatically disabled.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -3,69 +3,75 @@ on:
...
-      - name: Slack Notification
-        uses: rtCamp/action-slack-notify@v2
-        if: ${{ failure() }}
-        env:
-          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
-          SLACK_COLOR: red
-          SLACK_MESSAGE: 'Seems whatever you did failed :unrocket:'
-          SLACK_TITLE: Build failed
+      - name: Build and push Docker image
+        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
```
Comment (mb-shorai) [inline .github/workflows/main.yml]: I get that hashes are more secure, but on the other hand with tags we'd always get the latest actions versions e.g. with latest security patches (in actions themselves or their dependencies). 🤔

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,17 @@
-FROM registry.fedoraproject.org/fedora-minimal:35
+FROM rockylinux:9-minimal
 
-SHELL ["/bin/bash", "-l", "-c"]
+RUN microdnf install --nodocs -y crypto-policies-scripts && \
+    update-crypto-policies --set DEFAULT:SHA1 && \
```
Comment (mb-shorai) [inline Dockerfile:4]: :(

File: Dockerfile [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
 
 ENV npm_config_loglevel warn
 ENV npm_config_unsafe_perm true
 
-RUN microdnf --nodocs -y upgrade && \
-    microdnf --nodocs -y install fedora-repos-modular && \
-    microdnf module enable -y nodejs:${NODE_VERSION} && \
+RUN microdnf --nodocs -y upgrade && \    
+    curl -sL https://rpm.nodesource.com/pub_${NODEJS_VERSION}.x/nodistro/repo/nodesource-release-nodistro-1.noarch.rpm -o /tmp/nodejs.rpm && \
+    rpm -i /tmp/nodejs.rpm && \
```
Comment (mb-shorai) [inline Dockerfile]: rpm is able to install from a URL natively

File: docker-compose.yml [hunk 5]
```diff
@@ -1,15 +1,8 @@
...
-        NODE_VERSION: 14
-    image: nodejs:14
-
-  node16:
-    build:
-      context: .
-      args:
-        NODE_VERSION: 16
-    image: nodejs:16
+        NODE_VERSION: ${NODE_VERSION}
```
Comment (mb-shorai) [inline docker-compose.yml:7]: Perhaps set a default value

Comment (gothrek22) [reply docker-compose.yml:7]: But you can always run `NODE_VERSION=20` docker-compose build or something.

Comment (mb-shorai) [reply docker-compose.yml:7]: Yeah, doesn't matter, just nice to have

File: .github/workflows/main.yml (hunk 1 above)
Comment (gothrek22) [reply .github/workflows/main.yml:8]: Needs proper review of permissions, it's a checkbox for now as there is no risk of having too much. It's still tighter than it was weeks ago.

**Review by mb-shorai — APPROVED** [review]

**Pull request merged**